File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/auth/AuthViewModel.kt
package au.com.crazybean.mobilex.kurir.modules.auth

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import au.com.crazybean.mobilex.data.model.Auth
import au.com.crazybean.mobilex.modules.auth.AuthWrapper
import au.com.crazybean.sdk.mvvm.ViewModel

class AuthViewModel(private val wrapper: AuthWrapper) : ViewModel() {
    fun login(name: String, password: String): LiveData<Auth?> {
        return MutableLiveData<Auth?>().also { liveData ->
            wrapper.login(name, password) {
                liveData.value = it
            }
        }
    }
}

File: ios/Kurir/modules/auth/AuthViewModel.swift
//
//  AuthViewModel.swift
//  Kurir
//
//  Created by Loren on 19/8/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import MobileSDK
import Mobilex

class AuthViewModel: ViewModel {
    private let wrapper: AuthWrapper?
    
    init(_ wrapper: AuthWrapper?) {
        self.wrapper = wrapper
    }
    
    func login(name: String, password: String)-> LiveData<Auth?> {
        return LiveData<Auth?>() { [weak self] liveData in
            if let wrapper = self?.wrapper {
                wrapper.login(name: name, password: password, completion: { auth in
                    liveData.value = auth
                })
            }
        }
    }
}
